### ISSUE TITLE
Planter account duplicate detection, suggestion and merger

### ISSUE THREAD

**Issue by sebastiangaertner** [Greenstand/treetracker-admin#331]
Much like the contacts.google.com "Merge and fix" function it would be great to have a list of planters displayed in groups that share the same information such as device_id, telephone number and at least one name (exact parameters to be discussed). The grouped planter info would then be merged by an admin into one only when 100% sure about the duplicate account. The trees would also have to be reallocated to the new merged planter id. A great task for new developers.

**Comment by dadiorchen** [comment]
I have checked the Merge and fix function on contacts.google.com, that's a beautiful UX for this goal, I'd like to bring it into our admin panel :)

**Comment by BirdTho** [comment]
I think that on new account creation, the account details can be matched against other existing accounts, then set aside in a DB table for review with the matching entries. As they are merged, could either flag the duplicates in the duplicates table as resolved or  just delete them.
Do the merge, then flag the other account as inactive?

The point is, its cheap to check the account redundancy right at the creation instead of iterating through the account when we're   visiting that site. In fact, account creation can block you if email or phone is already in the system (unless for some reason there is a need to segregate an account for any reason, but I'd assume not). 

So two solutions - find potential matches at account creation, or block account creation due to email or phone being in use already.

**Comment by nmcharlton** [comment]
I agree with @BirdTho. Conflicts/potential merges could be surfaced on the Admin Panel, but the logic to detect them should sit elsewhere.

@sebastiangaertner Does the planter sign up workflow require internet connection, or can it be done offline and synced later? If the latter, this would make blocking account creation difficult, but checking for matches and flagging them on account creation would be a good idea.

@deepwinter Do we have any hooks/services that monitor changes in the DB? This might be a good way to handle merging, in case we want to update Planter details another way, such as through the Admin Panel.

**Comment by BirdTho** [comment]
An account that is created offline could be flagged as pending approval, these differences could be surfaced one connecting to the web and syncing a new account, and a message could be sent back that a user with these credentials already exists.

It would change the workflow but I think that rather than merging accounts it would be better not to create a new account in the first place if one pre-exists, and this is a standard across many online services. because accounts are not contacts like in Google contacts, they are accounts. They have different rules, especially in primary keys. You can make two identical contacts but you cannot make two identical accounts.

The only exception to this might be if someone makes an account with two different email addresses and the same phone number, they might be identified as duplicates. Maybe they forgot their login information for the first account and so they're trying to create a new one that they can access.
